improver_tests/blending/weighted_blend/test_WeightedBlendAcrossWholeDimension.py

```diff
@@ -753,6 +781,39 @@ def test_threshold_cube_with_weights_weighted_mean(self):
             result.coord("forecast_period").points, expected_forecast_period
         )
 
+    @ManageWarnings(ignored_messages=[COORD_COLLAPSE_WARNING])
+    def test_threshold_cube_with_promoted_dimension_weighted_mean(self):
```

Comment:
There's quite a lot of extra manipulation going on in this unit test. I wonder if you could do something simpler like this
```
    @ManageWarnings(ignored_messages=[COORD_COLLAPSE_WARNING])
    def test_threshold_cube_with_promoted_dimension_weighted_mean(self):
        """Test weighted_mean method works collapsing a cube with a threshold
        dimension when the blending is over a different coordinate. Note that
        this test is in process to include the slicing."""
        coord = "forecast_reference_time"
        plugin = WeightedBlendAcrossWholeDimension(coord)

        # Set up an input cube with one threshold as a dimension coordinate
        input_cube = self.cube_threshold[0]
        input_cube = iris.util.new_axis(input_cube, "precipitation_amount")

        result = plugin(input_cube, self.weights1d)

        expected_result_array = np.ones((1, 2, 2)) * 0.3
        expected_frt = int(self.cube.coord("forecast_reference_time").points[-1])
        expected_forecast_period = int(self.cube.coord("forecast_period").points[-1])

        self.assertArrayAlmostEqual(result.data, expected_result_array)
        self.assertEqual(result.attributes, self.expected_attributes)
        self.assertEqual(result.coord("time").points, expected_frt)
        self.assertEqual(
            result.coord("forecast_period").points, expected_forecast_period
        )
```

Reply:
Yours is nicer - I'll update the code with yours :-)